src/MARIGOLD/Condition.py

```diff
@@ -2470,73 +2611,688 @@ def calc_avg_lat_sep(self):
-        #f_TP         = 0.316*(1/(1-alpha_avg)/Rem)**0.25                # Two-phase frictional factor, Kong (2018)
-        f_TP         = 0.316*(mu_m_avg/mu_f/Rem)**0.25                # Two-phase frictional factor, Talley (2012) and Ted (2015)
-        eps          =  f_TP*v_m**3 /2/Dh                                # epsilon for calculating u_t
-            
-        for angle, r_dict in self.phi.items():
+
+        # Notes DHK 16 AUG
+        '''
+        Add a "use data" flag to toggle between using void reconstruct and local data to calculate covariance
+        Reconstructed profile should be saved in midas_dict
+
+        Verify mu_m_avg
+        '''
+
+        debug = True
+
+        if reconstruct_flag == True:
+            self.reconstruct_void(method='talley')
+            alpha_str = 'alpha_reconstructed'
+        else:
+            alpha_str = 'alpha'
+
+        alpha_avg       = self.area_avg(alpha_str)
+        ai_avg          = self.area_avg('ai')
+
+        rho_f           = self.rho_f                                        # Liquid phase density [kg/m**3]
+        rho_g           = self.rho_g                                        # Gas phase density [kg/m**3]
+        Dh              = self.Dh                                           # Hydraulic diameter
+        mu_f            = self.mu_f                                         # Dynamic viscosity of water [Pa-s]
+        mu_m            = mu_f / (1 - alpha_avg)
+
+        rho_m           = (1 - alpha_avg) * rho_f + alpha_avg * rho_g       # Mixture density
+        v_m             = (rho_f * self.jf + rho_g * self.jgloc) / rho_m    # Mixture velocity
+        Rem             = rho_m * v_m * Dh / mu_m                           # Mixture Reynolds number, ***CAREFUL*** I have seen some versions of the IATE script that use rho_f instead as an approximation
+        f_TP            = 0.316 * (mu_m / mu_f / Rem)**0.25                 # Two-phase friction factor, Talley (2012) and Worosz (2015), also used in iate_1d_1g
+        eps             = f_TP * v_m**3 / 2 / Dh                            # Energy dissipation rate (Wu et al., 1998; Kim, 1999), also used in iate_1d_1g
+        
+        if debug:
+            print(f"alpha_avg: {alpha_avg}")
+        
+        for angle, r_dict in self.data.items():
             for rstar, midas_dict in r_dict.items():
                 
-                if midas_dict['alpha'] <= alpha_cr:  # Check if local void fraction is less than or equal to alpha_cr
-                    u_t = 1.4 * eps**(1./3) * (midas_dict['Dsm1']/1000.)**(1./3) 
-                    # print(angle, rstar, Dh, v_m, Rem, f_TP, eps, midas_dict['Dsm1']) # for check
+                alpha_loc = midas_dict[alpha_str]
+                
+                if reconstruct_flag == True:
+                    ai_loc = ai_avg * alpha_loc / alpha_avg
+                    try:
+                        Db_loc = 6 * alpha_loc / ai_loc

```

Comment:
I guess it'd be a little causality loop. But you could base it off the average or something. Dsm1 doesn't change that much in my experience. But do what Talley did